client/material/button.tsx

```diff
@@ -307,6 +307,42 @@ export function useHotkeyHandler({
   return [setRefs, hotkeyHandler]
 }
 
+export interface HotkeyedLabelProps {
+  /** Whether the button is disabled (label will remain unchanged). */
+  disabled?: boolean
+  /** A hotkey that is registered for the button. The label will underline the hotkeyed letter. */
+  hotkey?: HotkeyProp
+  /** The given label of the button. If not a string, it will remain unchanged. */
+  label: string | React.ReactNode
+}
+
+export function useHotkeyedLabel({ disabled, hotkey, label }: HotkeyedLabelProps) {
```

Comment:
I'd prefer this was implemented as a component rather than a hook. As a general rule, try to avoid using JSX or returning elements in hooks. So phrased differently, if the thing needs to return an Element, it should be a component (or split between a hook + a component, but in this case no hook really needs to exist)

You can probably just memoize the component as well in this case, instead of doing `useMemo` inside of it